src/main/java/SuperPeer.java

```diff
@@ -1,55 +1,98 @@
  */
+import ether.TransactionsManager;
 import io.left.rightmesh.id.MeshID;
 import io.left.rightmesh.mesh.JavaMeshManager;
 import io.left.rightmesh.mesh.MeshManager;
-import io.left.rightmesh.mesh.MeshManager.MeshTransactionEvent;
-import io.left.rightmesh.util.MeshUtility;
+import io.left.rightmesh.mesh.MeshStateListener;
+import io.left.rightmesh.util.RightMeshException;
+import io.reactivex.functions.Consumer;
 
-import java.io.Console;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
 
+import static io.left.rightmesh.mesh.MeshManager.PEER_CHANGED;
 import static io.left.rightmesh.mesh.MeshManager.TRANSACTION_RECEIVED;
- 
-public class SuperPeer {
 
+public class SuperPeer implements MeshStateListener {
+
+    //TODO: Add logger instead of system.out
     public static final String TAG = SuperPeer.class.getCanonicalName();
-        
+
+    private boolean isRunning = true;
+    private TransactionsManager tm;
+
     public static void main(String[] args) {
         SuperPeer p = new SuperPeer();
     }
-    
+
     public SuperPeer() {
         JavaMeshManager mm = new JavaMeshManager(true);
-        mm.on(TRANSACTION_RECEIVED, this::handleTransactionPacket);
         System.out.println("Superpeer MeshID: " + mm.getUuid());
-        
+
         System.out.println("Waiting for lib to be ready");
         try {
-                Thread.sleep(200);
-        } catch(Exception ex) {}
-        
-        //test send on the TRANSACTION PORT
-        mm.sendDataReliable(mm.getUuid(), MeshUtility.TRANSACTION_PORT, "test".getBytes());
-        
-        //Continue to run until user quits
-        System.out.println("Enter empty line to quit");
-        Console c = System.console();
-        String msg;
-        if (c == null) {
-            System.err.println("No console.");
-            mm.stop();
-            return;
+            Thread.sleep(200);
+        } catch (Exception ex) {
         }
-        
+
+        tm = TransactionsManager.getInstance(mm);
+        tm.start();
+
+        String msg;
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+
         do {
-            msg = c.readLine();
-        } while(msg != null && !msg.equals(""));
-        
+            try {
+                msg = br.readLine();
+                processInput(msg);
+            } catch (IOException e) {
+                e.printStackTrace();
+                break;
+            }
+        } while (isRunning);
+
+        tm.stop();
         mm.stop();
     }
-    
-    public void handleTransactionPacket(MeshManager.RightMeshEvent e) {
-            System.out.println("TRANSACTION READY TO ROCK!");
-    }  
+
+    public void meshStateChanged(MeshID meshID, int state) {
+
+        if (state == MeshStateListener.SUCCESS) {
+            tm.start();
+        }
+    }
+
+    private void processInput(String msg) {
+        if(msg.equals("exit")) {
```

Comment:
Would make sense to make this a constant.

Reply:
done